executor/analyze.go

```diff
@@ -187,7 +188,11 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	}
 	if needGlobalStats {
 		for globalStatsID, info := range globalStatsMap {
-			globalStats, err := statsHandle.MergePartitionStats2GlobalStatsByTableID(e.ctx, e.opts, e.ctx.GetInfoSchema().(infoschema.InfoSchema), globalStatsID.tableID, info.isIndex, info.histIDs)
+			globalOpts := e.opts
+			if e.V2AnalyzeOptions != nil {
+				globalOpts = e.V2AnalyzeOptions.FilledOpts
```

Comment:
One `AnalyzeExec` may have multiple tables to analyze. For those different tables, though their explicitly-specified `OptsV2`s are the same, but their `FilledOpts` may be different since they may have different `tblSavedOpts`s. It is related to https://github.com/pingcap/tidb/pull/30939/files#r775116089

Reply:
Updated.